clouddriver-azure/src/main/groovy/com/netflix/spinnaker/clouddriver/azure/client/AzureStorageClient.groovy

```diff
@@ -16,21 +16,69 @@
 
 package com.netflix.spinnaker.clouddriver.azure.client
 
+import com.microsoft.azure.credentials.ApplicationTokenCredentials
+import com.microsoft.azure.management.storage.StorageManagementClient
+import com.microsoft.azure.management.storage.StorageManagementClientImpl
 import com.microsoft.azure.storage.blob.CloudBlobClient
 import com.microsoft.azure.storage.blob.CloudBlobContainer
 import com.microsoft.azure.storage.blob.CloudBlobDirectory
 import com.microsoft.azure.storage.blob.ListBlobItem
 import com.microsoft.azure.storage.CloudStorageAccount
+import com.microsoft.rest.ServiceResponse
 import com.netflix.spinnaker.clouddriver.azure.resources.vmimage.model.AzureCustomImageStorage
 import com.netflix.spinnaker.clouddriver.azure.resources.vmimage.model.AzureCustomVMImage
 import groovy.transform.CompileStatic
 import groovy.util.logging.Slf4j
+import okhttp3.logging.HttpLoggingInterceptor
 
 @Slf4j
 @CompileStatic
-class AzureStorageClient {
+class AzureStorageClient extends AzureBaseClient {
   static final String AZURE_IMAGE_FILE_EXT = ".vhd"
 
+  private final StorageManagementClient client
+
+  AzureStorageClient(String subscriptionId, StorageManagementClient client) {
+    super(subscriptionId)
+    this.client = client
+  }
+
+  AzureStorageClient(String subscriptionId, ApplicationTokenCredentials credentials) {
+    super(subscriptionId)
+    this.client = this.initialize(credentials)
+  }
+
+  /**
+   * get the StorageManagementClient which will be used for all interaction related to compute resources in Azure
+   * @param creds the credentials to use when communicating to the Azure subscription(s)
+   * @return an instance of the Azure StorageManagementClient
+   */
+  private StorageManagementClient initialize(ApplicationTokenCredentials tokenCredentials) {
+    StorageManagementClient storageClient = new StorageManagementClientImpl(tokenCredentials)
+    storageClient.setSubscriptionId(this.subscriptionId)
+    storageClient.setLogLevel(HttpLoggingInterceptor.Level.NONE)
```

Comment:
We should add a card on the board for this. There was talk about making the logging level something that could be changed on the fly (no restart required). Perhaps we could/should find a way to hook into that.